dune/stuff/test/grid_periodicview.cc

```diff
@@ -0,0 +1,295 @@
+        }
+      }
+      if (dimDomain == 2 && is_simplex) {
+        if (variant == 0) {
+          EXPECT_EQ(size_t(2*2*8*7 + 2*64),neighbor_count);
+          EXPECT_EQ(size_t(2*8*2), boundary_count);
+          EXPECT_EQ(size_t(0), periodic_count);
+        }
+        if (variant == 1) {
+          EXPECT_EQ(size_t(2*2*8*7 + 2*8 + 2*64),neighbor_count);
+          EXPECT_EQ(size_t(2*8*2 - 2*8), boundary_count);
+          EXPECT_EQ(size_t(2*8), periodic_count);
+        }
+        if (variant == 2) {
+          EXPECT_EQ(size_t(2*2*8*7 + 2*8*2 + 2*64),neighbor_count);
+          EXPECT_EQ(size_t(0), boundary_count);
+          EXPECT_EQ(size_t(2*8*2), periodic_count);
+        }
+      }
+      if (dimDomain == 3 && is_cube) {
+        if (variant == 0) {
+          EXPECT_EQ(size_t(2*3*7*64),neighbor_count);
+          EXPECT_EQ(size_t(6*64), boundary_count);
+          EXPECT_EQ(size_t(0), periodic_count);
+        }
+        if (variant == 1) {
+          EXPECT_EQ(size_t(2*3*7*64 + 2*64),neighbor_count);
+          EXPECT_EQ(size_t(6*64 - 2*64), boundary_count);
+          EXPECT_EQ(size_t(2*64), periodic_count);
+        }
+        if (variant == 2) {
+          EXPECT_EQ(size_t(2*3*7*64 + 6*64),neighbor_count);
+          EXPECT_EQ(size_t(0), boundary_count);
+          EXPECT_EQ(size_t(6*64), periodic_count);
+        }
+      }
+      if (dimDomain == 3 && is_simplex) {
+        if (variant == 0) {
+          EXPECT_EQ(size_t(2*2*3*7*64 + 2*6*8*8*8),neighbor_count);
+          EXPECT_EQ(size_t(2*6*64), boundary_count);
+          EXPECT_EQ(size_t(0), periodic_count);
+        }
+        if (variant == 1) {
+          EXPECT_EQ(size_t(2*2*3*7*64 + 2*2*64 + 2*6*8*8*8),neighbor_count);
+          EXPECT_EQ(size_t(2*6*64 - 2*2*64), boundary_count);
+          EXPECT_EQ(size_t(2*2*64), periodic_count);
+        }
+        if (variant == 2) {
+          EXPECT_EQ(size_t(2*2*3*7*64 + 2*6*64 + 2*6*8*8*8),neighbor_count);
+          EXPECT_EQ(size_t(0), boundary_count);
+          EXPECT_EQ(size_t(2*6*64), periodic_count);
+        }
+      }
+    }
+  }
+};
+
+template< class GridImp >
+struct PeriodicViewTestSimplex
+  : public PeriodicViewTestCube< GridImp >
+{};
+
+typedef testing::Types< YASPGRIDS
+# if HAVE_ALUGRID
+                      , ALUCUBEGRIDS
+# endif
+                      > CubeGridTypes;
+
+typedef testing::Types<
+# if HAVE_ALUGRID
+                        ALUSIMPLEXGRIDS
+# endif
+                      > SimplexGridTypes;
+
+
+TYPED_TEST_CASE(PeriodicViewTestCube, CubeGridTypes);
+TYPED_TEST(PeriodicViewTestCube, checkcube)
+{
+  this->check(false);
+}
+
+TYPED_TEST_CASE(PeriodicViewTestSimplex, SimplexGridTypes);
+TYPED_TEST(PeriodicViewTestSimplex, checksimplex)
+{
+  this->check(true);
+}
+
+
+#else // HAVE_DUNE_GRID
+
+TEST(DISABLED_CubeGridProvider, is_default_creatable) {}
+TEST(DISABLED_CubeGridProvider, fulfills_const_interface) {}
+TEST(DISABLED_CubeGridProvider, is_visualizable) {}
```

Comment:
These have to be adapted to your tests, e.g, `TEST(DISABLED_PeriodicViewTestCube, checkcube) {};`.